packages/react-storage/src/components/StorageBrowser/Views/Controls/Navigate.tsx

```diff
@@ -93,23 +88,53 @@ export interface NavigateControl<
 }
 
 export const NavigateControl: NavigateControl = (_props) => {
+  const [state, handleUpdateState] = useControl({ type: 'NAVIGATE' });
+
   return (
     <NavigateContainer>
-      <NavigateItem
-        item={{
-          name: 'Home',
-          permissionType: 'READ',
-          locationType: 'OBJECT',
-        }}
-      />
-      <NavigateItem
-        item={{
-          name: 'Location01',
-          permissionType: 'READ',
-          locationType: 'OBJECT',
-        }}
-        current
-      />
+      <NavigateItemContainer>
+        <NavigateButton
+          onClick={() => handleUpdateState({ type: 'DESELECT_LOCATION' })}
+        >
+          Home
+        </NavigateButton>
+        {state.location ? <Separator /> : null}
+      </NavigateItemContainer>
+      {state.location.current && (
+        <NavigateItemContainer>
+          <NavigateButton
+            onClick={() => {
+              if (state?.location.current) {
+                handleUpdateState({
+                  type: 'SELECT_LOCATION',
+                  location: state.location.current,
+                });
+              }
+            }}
+          >
+            {state.location.current.bucket}
+          </NavigateButton>
+          {state.history.list && state.history.list.length > 0 ? (
+            <Separator />
+          ) : null}
+        </NavigateItemContainer>
+      )}
+      {state.history &&
+        state.history.list?.map((folder, index) => (
+          <NavigateItem
+            key={index}
+            item={folder}
+            current={
+              folder === state.history.list?.[state.history.list.length - 1]
+            }
+            onClick={() => {
+              handleUpdateState({
+                type: 'EXIT_FOLDER',
+                index,
+              });
+            }}
```

Comment:
Let's sync on behavior here, if we need to pass something here it should probably be the `folder`

Reply:
Sounds good. I might be misunderstanding how the `Navigate` control should be working. I thought it was supposed to work like breadcrumbs and so I think it needs a way to navigate to a specific folder that isn't on the top of the stack.

For instance: Home / bucket / "folder1" / "folder2" / "folder3", if I wanted to go back to "folder1", should that be allowed or are we only going back up one folder at a time?